examples/classic_controllers/classic_controllers.py

```diff
@@ -688,41 +902,507 @@ def control(self, state, reference):
             for i in range(3):
                 action += (2 ** (2 - i)) * self.abc_controller[i].control(state[self.i_abc_idx[i]], ref_abc[i])
 
+        self.plot(self.external_ref_plots, self.state_names)
         return action
 
-    def plot(self):
-        return dict(ref_state=self.ref_state_idx[:-1], ref_value=self.ref[:-1], color=self.color)
+    def set_ref(self):
+        return dict(ref_state=self.ref_state_idx[:-1], ref_value=self.ref[:-1])
 
     def reset(self):
-        self.overlayed_controller.reset()
+        for overlayed_controller in self.overlayed_controller:
+            overlayed_controller.reset()
         if self.controller_type:
             self.d_controller.reset()
-            for q_controller in self.q_controller:
-                q_controller.reset()
+            self.q_controller.reset()
+
         else:
             for abc_controller in self.abc_controller:
                 abc_controller.reset()
 
+        if self.torque_control:
+            self.torque_controller.reset()
+
+class MTPC:
+    """
+        This class represents the torque controller for cascaded control of synchronous motors.  For low speeds only the
+        current limitation of the motor is important. The current vector to set a desired torque is selected so that the
+        amount of the current vector is minimum (Maximum Torque per Current). For higher speeds, the voltage limitation
+        of the synchronous motor or the actuator must also be taken into account. This is done by converting the available
+        voltage to a speed-dependent maximum flux. An additional modulation controller is used for the flux control. By
+        limiting the flux and the maximum torque per flux (MTPF), an operating point for the flux and the torque is
+        obtained. This is then converted into a current operating point. The conversion can be done by different methods
+        (parameter torque_control). On the one hand, maps can be determined in advance by interpolation or analytically,
+        or the analytical determination can be done online.
+        For the visualization of the operating points, both for the current operating points as well as the flux and
+        torque operating points, predefined plots are available (plot_torque: default True). Also the values of the
+        modulation controller can be visualized (plot_modulation: default False).
+    """
+
+    def __init__(self, environment, plot_torque=True, plot_modulation=False, update_interval=1000, torque_control='interpolate'):
+        self.mp = environment.physical_system.electrical_motor.motor_parameter
+        self.limit = environment.physical_system.limits
+        self.nominal_values = environment.physical_system.nominal_state
+        self.torque_control = torque_control
+
+        self.q = environment.physical_system.electrical_motor.q
+        self.a_max = 2 / np.sqrt(3)
+        self.k_ = 0.95
+        d = 1.2
+        alpha = d / (d - np.sqrt(d ** 2 - 1))
```

Comment:
These variables (and their assigned values) are probably not very self-explanatory. They could use some in-line commenting and / or more descriptive names.

Reply:
I have added in-line comments.